### PR TITLE
feat: Add REUSE license plugin to extract license information for built assets

### PR DESCRIPTION
**Pull request by susnux**
This is a vite version of the plugin we use on Nextcloud server for Webpack to extract the license information.

(I still think it makes sense to externalize the plugins a bit a allow others to reuse them :sweat_smile: )

### CODE REVIEW

**Comment by susnux** [comment]
cc @AndyScherzinger this allows then to also remove the .dep5 stuff for e.g. activity

File: lib/appConfig.ts [hunk 1]
```diff
@@ -71,6 +72,12 @@ export interface AppOptions extends Omit<BaseOptions, 'inlineCSS'> {
 	 * @default 'js/vendor.LICENSE.txt'
 	 */
 	thirdPartyLicense?: false | string
+
+	/**
+	 * Extract license information from built assets into `.license` files
+	 * This is needed to be REUSE complient
```
Comment (AndyScherzinger) [inline lib/appConfig.ts:78]: complient -> compliant

Comment (susnux) [reply lib/appConfig.ts:78]:
```suggestion
	 * This is needed to be REUSE compliant
```

**Review by AndyScherzinger — APPROVED** [review]

File: lib/appConfig.ts [hunk 2]
```diff
@@ -145,14 +152,22 @@ export const createAppConfig = (entries: { [entryAlias: string]: string }, optio
 				plugins.push(CSSEntryPointsPlugin({ createEmptyEntryPoints: options.createEmptyCSSEntryPoints }))
 			}
 
+			if (options.extractLicenseInformation) {
+				plugins.push(REUSELicensesPlugin(
+					typeof options.extractLicenseInformation === 'object'
+						? options.extractLicenseInformation
+						: {},
+				))
+			}
```
Comment (ShGKme) [inline lib/appConfig.ts:161]: Should we enable it only in production build?

Comment (susnux) [reply lib/appConfig.ts:161]: Sure makes sense!

**Pull request merged**